src/Castle.Core/Components.DictionaryAdapter/Xml/Internal/Utilities/SingletonDispenser.cs

```diff
@@ -18,12 +18,11 @@ namespace Castle.Components.DictionaryAdapter.Xml
 	using System;
 	using System.Collections.Generic;
 	using System.Threading;
-	using Castle.Core.Internal;
 
 	public class SingletonDispenser<TKey, TItem>
```

Comment:
If this is an internal utility why is it public? Should it be [Obsolete] too?

Reply:
Or it should be `internal`, like as in the csharp modifier we all know and love :)